utils/s2n_safety.h

```diff
@@ -99,9 +99,6 @@ static inline void* trace_memcpy_check(void *restrict to, const void *restrict f
 #define GUARD_NONNULL_GOTO( x , label ) do {if ( (x) == NULL ) goto label;} while (0)
 #define GUARD_NONNULL_PTR( x )          do {if ( (x) == NULL ) return NULL;} while (0)
 
-/* Check the return value from caller. If this value is -2, S2N_ERR_BLOCKED is marked*/
-#define GUARD_AGAIN( x )  do {if ( (x) == -2 ) { S2N_ERROR(S2N_ERR_BLOCKED); } GUARD( x );} while(0)
```

Comment:
I removed it originally because I thought caching was a one-off design. I agree it should be put back because there will be multiple message handlers that need to be non-blocking.